src/nvim/tui/term_input.inl

```diff
@@ -257,19 +258,17 @@ static TermInput *term_input_new(void)
   rv->paste_enabled = false;
   rv->in_fd = 0;
 
-  // Set terminal encoding based on environment(taken from libtermkey source
-  // code)
-  const char *e;
-  int flags = 0;
-  if (((e = os_getenv("LANG")) || (e = os_getenv("LC_MESSAGES"))
-        || (e = os_getenv("LC_ALL"))) && (e = strchr(e, '.')) && e++ &&
-      (strcasecmp(e, "UTF-8") == 0 || strcasecmp(e, "UTF8") == 0)) {
-    flags |= TERMKEY_FLAG_UTF8;
+  // Set terminal encoding based on environment
+  int enc_flag;
+  char *enc = (char *)enc_locale();
```

Comment:
`set_init_1()` checks this very early, and then tests `mb_init()` to see if the result is valid:

https://github.com/neovim/neovim/blob/98b3dc0571603f07f3e81babe402bb0837fc619b/src/nvim/option.c#L1943

If mb_init() fails, we default `p_enc` to utf-8. Is there any reason we wouldn't want to use `p_enc`? Is it conceivable that a user would `set encoding=...`, and yet want something *else* for the terminal encoding?

Actually, based on the docstring for `mb_init()` it looks like you may be able to just check the `enc_utf8` flag:

    rv->tk = termkey_new_abstract(os_getenv("TERM"), 
      enc_utf8 ? TERMKEY_FLAG_UTF8 : TERMKEY_FLAG_RAW);